R/colby_constructors.R

```diff
@@ -871,11 +883,62 @@ split_rows_by_cutfun <- function(lyt, var,
 #' subsets of the data. These numbers will not, and currently cannot, reflect
 #' alternate column observation counts provided by the `alt_counts_df`,
 #' `col_counts` or `col_total` arguments to \code{\link{build_table}}}
+#' 
+#' @name spl_context
 NULL
 
-
-
-
+#' @title Additional parameters within analysis and content functions 
+#' (`afun/cfun`)
+#' 
+#' @description 
+#' It is possible to add specific parameters to `afun` and `cfun`, in [analyze]
+#' and [summarize_row_groups] respectively. These parameters grant access to 
+#' relevant information like the row split structure (see [spl_context]) and the
+#' predefined baseline (`.ref_group`).
+#' 
+#' @details
+#' We list and describe here all the parameters that can be added to a custom 
+#' analysis function:
+#' 
+#' \describe{
+#'   \item{.N_col}{column-wise N (column count) for the full column being
+#'     tabulated within}
+#'   \item{.N_total}{overall N (all observation count, defined as sum of column
+#'     counts) for the tabulation}
+#'   \item{.N_row}{row-wise N (row group count) for the group of observations
+#'     being analyzed (ie with no column-based subsetting)}
+#'   \item{.df_row}{data.frame for observations in the row group being analyzed
+#'     (ie with no column-based subsetting)}
+#'   \item{.var}{variable that is analyzed}
+#'   \item{.ref_group}{data.frame or vector of subset corresponding to the
+#'     `ref_group` column including subsetting defined by row-splitting.
+#'     Optional and only required/meaningful if a `ref_group` column has been
+#'     defined}
+#'   \item{.ref_full}{data.frame or vector of subset corresponding to the
+#'     `ref_group` column without subsetting defined by row-splitting. Optional
+#'     and only required/meaningful if a `ref_group` column has been defined}
+#'   \item{.in_ref_col}{boolean indicates if calculation is done for cells
+#'     within the reference column}
+#'   \item{.spl_context}{data.frame, each row gives information about a
+#'     previous/'ancestor' split state. See \code{\link{spl_context}}}
+#'   \item{.alt_counts_df}{list of data.frames, i.e. the `alt_count_df` after 
+#'     row splitting. Each element of the list represent a column-wise split.}
```

Comment:
Currently `.alt_counts_df` is a single data.frame and not a list. Is there a use case for this argument? The naming is confusing since `.alt_counts_df` gives the `alt_counts_df` after the row split but typically `alt_counts_df` is typically related to the column N's and column splits.

Reply:
Yes, you are right. I will change the doc. For a bit of context (pun not intended), in the beginning, I thought to have `.alt_df` see all the other row splits and not only the current one but it is out of scope for what Emily is using it and still, if this is the case, it would go into `.spl_context`. 

So there are two levels of faceting/subsetting here that someone might want to access: 1. all the same level facets, 2. all the hierarchical splits facets (currently what `full_df` is doing in `.spl_context`). The expected use needs neither, but only the current row split data frame. This, in turn, needs to be subsetted by different column expressions (hence all_col_exprs) for getting the right statistics (it is necessary for a case in which there is a combo df)